builds/jenkins-build-pack-api/Jenkinsfile

```diff
@@ -830,41 +861,45 @@ def updateOperationId(swaggerFileContent) {
 def writeServerlessFile(config, environment_logical_id, accountDetails){
 
   def iamRoleArnValue;
-  if(config['domain'] == 'jazz'){
-    iamRoleArnValue = config['iamRoleARN']
-  } else {
-    iamRoleArnValue = accountDetails.IAM.USERSERVICES_ROLEID;
-  }
+  if(config['domain'] == 'jazz') iamRoleArnValue = config['iamRoleARN']
```

Comment:
This has to be updated a little bit. 
- `accountDetails.IAM.USERSERVICES_ROLEID` will never be used in the workflows again. We either create a dedicated role per user service/env & used it OR just use user provided role.
- Why is `iamRoleArnValue = config['iamRoleARN']` only  when `domain == jazz`? It should be true for all services from now on. If user provides custom role, we have it use it during safe updates and deployments

Reply:
See my comments https://github.com/tmobile/jazz/pull/1227/files?utf8=%E2%9C%93&diff=unified&w=1#r351057997 on why this works if vault is enabled/disabled.

Reply:
In oss platform services(jazz core) have elevated permissions hence iamRoleArnValue = config['iamRoleARN'] only when domain == jazz

Reply:
@SiniWilson @devsatishm This change is pending support for dedicated roles per environment.